### PR TITLE
[ETL-156, ETL-157, ETL-158] Create configuration dictionary for GENIE code

### PR DESCRIPTION
**Pull request by thomasyu888**
- create get genie configuration function
- Utilize genie configuration dict for all validation functions

### CODE REVIEW

**Comment by thomasyu888** [comment]
@hhunterzinck this is ready to be reviewed.  Sorry for the slightly large PR

File: bin/input_to_database.py [hunk 1]
```diff
@@ -26,69 +31,86 @@ def main(
...
-    databaseToSynIdMapping = syn.tableQuery(
-        "SELECT * FROM {}".format(database_to_synid_mapping_synid[0])
-    )
-    databaseToSynIdMappingDf = databaseToSynIdMapping.asDataFrame()
+    # Get project GENIE configurations
+    database_to_synid_mapping_synid = project.annotations.get("dbMapping", "")
 
-    center_mapping_id = process_functions.getDatabaseSynId(
-        syn, "centerMapping", databaseToSynIdMappingDf=databaseToSynIdMappingDf
+    # TODO: remove this once code is using genie config
```
Comment (hhunterzinck) [inline bin/input_to_database.py]: As in, incorporating genie_config into the genie_registry scripts for each file type?

File: bin/input_to_database.py [hunk 2]
```diff
@@ -26,69 +31,86 @@ def main(
...
-        #                  'UHN', 'MDA', 'WAKE', 'YALE', 'UCSF', 'CRUK',
-        #                  'CHOP', 'VHIO', 'SCI', 'PHS', 'COLU', 'UCHI']
-        center_mapping_df = center_mapping_df[~center_mapping_df["inputSynId"].isnull()]
-        # release is a bool column
-        center_mapping_df = center_mapping_df[center_mapping_df["release"]]
-        # center_mapping_df = center_mapping_df[
-        #     ~center_mapping_df['center'].isin(exclude_sites)
-        # ]
-        centers = center_mapping_df.center
+        # TODO: add in logic to exclude sites from processing
```
Comment (hhunterzinck) [inline bin/input_to_database.py:71]: Is there a reason not to add this as another element in the `center_config` object?

File: bin/input_to_database.py [hunk 3]
```diff
@@ -26,69 +31,86 @@ def main(
...
     if oncotree_link is None:
-        onco_link = databaseToSynIdMappingDf["Id"][
-            databaseToSynIdMappingDf["Database"] == "oncotreeLink"
-        ].values[0]
-        onco_link_ent = syn.get(onco_link)
+        onco_link_ent = syn.get(genie_config["oncotreeLink"])
         oncotree_link = onco_link_ent.externalURL
+        genie_config["oncotreeLink"] = onco_link_ent.externalURL
+    else:
+        genie_config["oncotreeLink"] = oncotree_link
```
Comment (hhunterzinck) [inline bin/input_to_database.py]:
```suggestion
        onco_link_ent = syn.get(genie_config["oncotreeLink"])
        oncotree_link = onco_link_ent.externalURL
    genie_config["oncotreeLink"] = oncotree_link
```

File: bin/input_to_database.py [hunk 4]
```diff
@@ -26,69 +31,86 @@ def main(
...
     # Check if you can connect to oncotree link,
     # if not then don't run validation / processing
-    process_functions.checkUrl(oncotree_link)
+    process_functions.checkUrl(genie_config["oncotreeLink"])
```
Comment (hhunterzinck) [inline bin/input_to_database.py:81]: This is great

File: bin/input_to_database.py [hunk 5]
```diff
@@ -26,69 +31,86 @@ def main(
...
+
+    # HACK: Add genie annotation package to config
+    genie_config["genie_annotation_pkg"] = genie_annotation_pkg
```
Comment (hhunterzinck) [inline bin/input_to_database.py:86]: What happens when the user doesn't specify the package given the default is None?

File: bin/input_to_database.py [hunk 6]
```diff
@@ -26,69 +31,86 @@ def main(
...
 
-    center_mapping_ent = syn.get(center_mapping_id)
+    # HACK: Not sure if this is needed anymore
```
Comment (hhunterzinck) [inline bin/input_to_database.py]: this is set to false again on line 145 and I don't see any references to the `isProcessing` in python scripts in `bin`, `genie`, or `genie_registry`

File: genie_registry/clinical.py [hunk 7]
```diff
@@ -493,18 +487,16 @@ def _validate(self, clinicaldf, oncotree_link):
         clinicaldf = clinicaldf.fillna("")
 
         oncotree_mapping_dict = process_functions.get_oncotree_code_mappings(
-            oncotree_link
+            self.genie_config["oncotreeLink"]
```
Comment (hhunterzinck) [inline genie_registry/clinical.py:496]: Is `genie_config` being passed to file objects upon instantiation?  When running the command 
`genie validate --project_id syn7208886 data_clinical_supp_SAGE.txt SAGE`

I get the following error
 

> Traceback (most recent call last):
>   File "/Users/hhunterzinck/opt/anaconda3/envs/genie/bin/genie", line 8, in <module>
>     sys.exit(main())
>   File "/Users/hhunterzinck/opt/anaconda3/envs/Genie/lib/python3.8/site-packages/genie/__main__.py", line 176, in main
>     args.func(syn, args)
>   File "/Users/hhunterzinck/opt/anaconda3/envs/Genie/lib/python3.8/site-packages/genie/validate.py", line 263, in _perform_validate
>     valid, message = validator.validate_single_file(**mykwargs)
>   File "/Users/hhunterzinck/opt/anaconda3/envs/Genie/lib/python3.8/site-packages/genie/validate.py", line 105, in validate_single_file
>     valid, errors, warnings = validator.validate(
>   File "/Users/hhunterzinck/opt/anaconda3/envs/Genie/lib/python3.8/site-packages/genie/example_filetype_format.py", line 178, in validate
>     errors, warnings = self._validate(df, **mykwargs)
>   File "/Users/hhunterzinck/opt/anaconda3/envs/Genie/lib/python3.8/site-packages/genie_registry/clinical.py", line 490, in _validate
>     self.genie_config["oncotreeLink"]
> TypeError: 'NoneType' object is not subscriptable
>

File: genie/input_to_database.py [hunk 8]
```diff
@@ -747,28 +726,45 @@ def validation(
...
+        process (str): _description_
+        only_validate (bool): _description_
+        database_to_synid_mappingdf (pd.DataFrame): _description_
+        center_mapping_df (pd.DataFrame): _description_
+        delete_old (bool, optional): _description_. Defaults to False.
+        oncotree_link (str, optional): _description_. Defaults to None.
+        genie_annotation_pkg (str, optional): _description_. Defaults to None.
+        format_registry (typing.List, optional): _description_. Defaults to None.
+        genie_config (typing.Dict, optional): See example of genie config at
+                                              ./genie_config.json. Defaults to None.
```
Comment (hhunterzinck) [inline genie/input_to_database.py:750]: Where is the `genie_config.json` example located?  I couldn't find any json files in the repo

**Review by hhunterzinck — CHANGES_REQUESTED** [review]
There was only one real issue that I found with the `genie_config` dictionary when trying to run on a test file.  The rest of the comments are mostly questions to confirm my understanding.

File: bin/input_to_database.py (hunk 5 above)
Comment (thomasyu888) [reply bin/input_to_database.py:86]: I added a check for this in the other PR.  Its only required when the 'mutation' processing is passed in.

File: bin/input_to_database.py (hunk 6 above)
Comment (thomasyu888) [reply bin/input_to_database.py]: This is explained in the other PR as well

File: genie_registry/clinical.py (hunk 7 above)
Comment (thomasyu888) [reply genie_registry/clinical.py:496]: This is an error on my part. Need to add to the validate cli.

File: genie/input_to_database.py (hunk 8 above)
Comment (thomasyu888) [reply genie/input_to_database.py:750]: It's not actually. I need to include an example but you can currently find it at tests/conftest.py.  (unless that's in the other PR as well)

File: bin/input_to_database.py (hunk 2 above)
Comment (thomasyu888) [reply bin/input_to_database.py:71]: There isn't but that's another ticket

File: bin/input_to_database.py (hunk 1 above)
Comment (thomasyu888) [reply bin/input_to_database.py]: I am. As part of each class, I'm now passing in the genie config.  There should be an edit here on the example_file_format class

**Review by hhunterzinck — APPROVED** [review]

**Pull request merged**